### ISSUE TITLE
Psycopg3 support

### ISSUE THREAD

**Issue by NathanSmeltzer** [PaulGilmartin/django-pgpubsub#56]
Are there plans to support Psycopg 3?
I tried installing both psycopg2 and psycopg[c] (Psycopg3) but received this error which seems due to Psycopg 3 not having a poll method for its Connection class:
```
File "/usr/local/lib/python3.10/site-packages/pgpubsub/listen.py", line 71, in process_notifications
    pg_connection.poll()
AttributeError: 'Connection' object has no attribute 'poll'
```

Supporting this would clear up some confusion as cookiecutter django now installs Psycopg 3 by default. Libraries like django-debug-toolbar [support both versions](https://github.com/jazzband/django-debug-toolbar/issues/1736).

**Comment by PaulGilmartin** [comment]
Yes I will try to support this in the next version of this library. I haven't worked with psycopg3 yet, so I'm not sure of the work involved and how practical it will be to support both 2 and 3 on the same version, but I will definitely look into it for the next release of this library (hoping that's sometime at the end of summer).

**Comment by romank0** [comment]
> @romank0 I know you're using a forked version of this library. Have you tried it with psycopg3 yet at all?

No, we are still using v2.

**Comment by jayjlow** [comment]
I made some alterations to the process_notifications function:

`def process_notifications(pg_connection):
    # pg_connection.poll()
    gen = pg_connection.notifies()
    for notification in gen:
        with transaction.atomic():
            for processor in [
                NotificationProcessor,
                LockableNotificationProcessor,
                NotificationRecoveryProcessor,
            ]:
                try:
                    processor = processor(notification, pg_connection)
                except InvalidNotificationProcessor:
                    continue
                else:
                    processor.process()
                    break
        if notification.payload == "stop":
            gen.close()`

Not sure if there is more to do, but this at least got the notify working for me.

**Comment by PaulGilmartin** [comment]
@jayjlow Thanks for the tip! I'll take that into account when I return to this task.

**Comment by PaulGilmartin** [comment]
Support for psycogp3 has been implemented in v1.3.0 of this library thanks to @romank0. See the [installation docs](https://django-pgpubsub.readthedocs.io/en/latest/installation.html) for more details. Note that when using psycogp3, we need to perform a query to fetch the notifications when listening. This is due to differences in the library which currently have made it impossible (as far as we can see) to do it client side like on psycopg2.
